### PR TITLE
Convert cli and treeherder to pytest

### PR DESCRIPTION
**Pull request by rgpt**
* Convert Cli and treeherder to pytest
* separating Class in a separate File
* Pylint Check Done on the files.

### CODE REVIEW

**Comment by rgpt** [comment]
@whimboo PR added. Please Review.

File: tests/cli/test_cli_arguments.py [hunk 1]
```diff
@@ -5,17 +5,13 @@
 # You can obtain one at http://mozilla.org/MPL/2.0/.
 
 import subprocess
-import unittest
+import re
 
-
-class TestCLIArguments(unittest.TestCase):
+def test_unrecognized_argument():
     """Test CLI entry script for arguments."""
```
Comment (whimboo) [inline tests/cli/test_cli_arguments.py]: You can remove this doc string. The test name is clear enough. It also applies to other methods below.

File: tests/cli/test_correct_scraper.py [hunk 2]
```diff
@@ -40,25 +38,22 @@
         'fname': '8fcac92cfcad-firefox-38.0a1.en-US.win32.installer.exe',
     },
 }
-
-
-class TestCLICorrectScraper(mhttpd.MozHttpdBaseTest):
+@pytest.mark.parametrize("data", tests.values())
+def test_cli_scraper(httpd, tmpdir, data, mocker):
```
Comment (whimboo) [inline tests/cli/test_correct_scraper.py]: `test_correct_cli_scraper` please.

File: tests/cli/test_correct_scraper.py [hunk 3]
```diff
@@ -40,25 +38,22 @@
...
+    args.extend(data['args'])
+    cli.cli(args)
 
-            dir_content = os.listdir(self.temp_dir)
-            self.assertTrue(data['fname'] in dir_content)
+    dir_content = os.listdir(str(tmpdir))
+    assert data['fname'] in dir_content
 
-            mozfile.remove(os.path.join(self.temp_dir, data['fname']))
+    mozfile.remove(os.path.join(str(tmpdir), data['fname']))
```
Comment (whimboo) [inline tests/cli/test_correct_scraper.py]: This line can be removed because pytest removes the tmpdir itself.

**Comment by rgpt** [comment]
@whimboo  PR updated. Please review. After this i will see to it if we can remove mozfile dependencies.

File: tests/cli/test_correct_scraper.py [hunk 4]
```diff
@@ -7,11 +7,9 @@
 import os
 
 import mozfile
```
Comment (whimboo) [inline tests/cli/test_correct_scraper.py]: A left-over import for mozfile. Can you please remove it?

**Comment by rgpt** [comment]
@whimboo PR Updated. Please Check.

**Comment by whimboo** [comment]
Thanks a lot! It's great to see that we can sign-off for two more sub directories!

**Pull request merged**